modules/client-handler/src/main/java/org/apache/ignite/client/handler/ClientHandlerModule.java

```diff
@@ -319,32 +313,35 @@ protected void initChannel(Channel ch) {
                 .option(ChannelOption.CONNECT_TIMEOUT_MILLIS, configuration.connectTimeout());
 
         int port = configuration.port();
-        Channel ch = null;
-
-        ChannelFuture bindRes = bootstrap.bind(port).await();
-
-        if (bindRes.isSuccess()) {
-            ch = bindRes.channel();
-        } else if (!(bindRes.cause() instanceof BindException)) {
-            throw new IgniteException(
-                    INTERNAL_ERR,
-                    "Failed to start thin client connector endpoint: " + bindRes.cause().getMessage(),
-                    bindRes.cause());
-        }
-
-        if (ch == null) {
-            String msg = "Cannot start thin client connector endpoint. Port " + port + " is in use.";
-
-            LOG.debug(msg);
-
-            throw new IgniteException(ErrorGroups.Network.PORT_IN_USE_ERR, msg);
-        }
-
-        if (LOG.isInfoEnabled()) {
-            LOG.info("Thin client protocol started successfully [port={}]", port);
-        }
 
-        return ch.closeFuture();
+        bootstrap.bind(port).addListener((ChannelFutureListener) bindFut -> {
+            if (bindFut.isSuccess()) {
+                if (LOG.isInfoEnabled()) {
+                    LOG.info("Thin client protocol started successfully [port={}]", port);
```

Comment:
Sounds a little weird, does not it?

Reply:
Agree, fixed to be consistent with other messages below.